src/server.c

```diff
@@ -2602,6 +2614,7 @@ void initDbState(redisDb *db){
         db->sub_dict[subdict].key_count = 0;
         db->sub_dict[subdict].resize_cursor = 0;
         db->sub_dict[subdict].slot_size_index = server.cluster_enabled ? zcalloc(sizeof(unsigned long long) * (CLUSTER_SLOTS + 1)) : NULL;
+        db->sub_dict[subdict].bucket_count = (CLUSTER_SLOTS * DICT_HT_INITIAL_SIZE);
```

Comment:
maybe we can use dictBuckets? instead of the internal DICT_HT_INITIAL_SIZE?

Reply:
isn't that broken in non-cluster mode?
i.e. we rely on this variable in that mode too, but the logic here refers to CLUSTER_SLOTS.

Reply:
also, shouldn't this mechanism skip non-local slot (for which we don't hold a dict, right?)?

Reply:
Addressed these in the recent commit. PTAL